tfx/components/example_gen/utils.py

```diff
@@ -300,46 +367,60 @@ def _retrieve_latest_span_version(
             'Cannot find %s number from %s based on %s' %
             (VERSION_PROPERTY_NAME, file_path, split_regex_pattern))
 
-    if latest_span is None or span_int > int(latest_span):
+    if latest_span_int is None or span_int > latest_span_int:
       # Uses str instead of int because of zero padding digits.
-      latest_span = span_str
+      latest_span_elems = span_strs
+      latest_span_int = span_int
       latest_version = version_str
-    elif (span_int == int(latest_span) and
+    elif (latest_span_int == span_int and
           (latest_version is None or version_int >= int(latest_version))):
       latest_version = version_str
 
-  if latest_span is None or (is_match_version and latest_version is None):
+  if latest_span_int is None or (is_match_version and
+                                 latest_version is None):
     raise ValueError('Cannot find matching for split %s based on %s' %
                      (split.name, split.pattern))
 
-  return latest_span, latest_version
+  # Update split pattern so executor can find the files to ingest.
+  if is_match_span:
+    split.pattern = split.pattern.replace(SPAN_SPEC, latest_span_elems[0])
+  elif is_match_date:
+    for spec, value in zip(DATE_SPECS, latest_span_elems):
+      split.pattern = split.pattern.replace(spec, value)
+
+  latest_version_int = None
+  if is_match_version:
+    split.pattern = split.pattern.replace(VERSION_SPEC, latest_version)
+    latest_version_int = int(latest_version)
+
+  return latest_span_int, latest_version_int
```

Comment:
nit, the function still a bit long, probably decompose it to logical steps improve readability, 

e.g., 
_find_matched_span_tokens_from_path
_resolve_latestest_span_version

Reply:
addressed. decomposed span matching as well as glob and regex generation to two different private functions.